src/span_context.js

```diff
@@ -0,0 +1,163 @@
+// @flow
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import * as constants from './constants.js';
+import Utils from './util.js';
+import Int64 from 'node-int64';
+
+export default class SpanContext {
+    _traceId: any;
+    _spanId: any;
+    _parentId: any;
+    _flags: number;
+    _baggage: any;
+    _baggageHeaderCache: any;
+
+    constructor(traceId: any,
+                spanId: any,
+                parentId: any,
+                flags: number) {
+        this._traceId = traceId;
+        this._spanId = spanId;
+        this._parentId = parentId;
+        this._flags = flags;
+        this._baggage = {};
+        this._baggageHeaderCache = {};
```

Comment:
cache only makes sense as static cache, but this looks like an instance variable, which will be unique for each instance of SpanContext